src/pages/SubmissionConfig.js

```diff
@@ -97,19 +130,45 @@ export const SubmissionConfig = ({ methodId }) => {
     }
   }
 
-  const loadTablesData = async () => {
+  const loadRecordTypes = useCallback(async wdsUrlRoot => {
     try {
-      setRecordTypes(await Ajax(signal).Wds.types.get())
+      setRecordTypes(await Ajax(signal).Wds.types.get(wdsUrlRoot))
     } catch (error) {
-      notify('error', 'Error loading tables data', { detail: await (error instanceof Response ? error.text() : error) })
+      notify('error', 'Error loading data types', { detail: await (error instanceof Response ? error.text() : error) })
     }
-  }
+  }, [signal])
+
+  const loadWdsData = useCallback(async (recordType, isLoadRecordTypes) => {
+    try {
+      // try to load WDS proxy URL if one doesn't exist
+      if (!wdsProxyUrl || (wdsProxyUrl.status !== 'Ready')) {
+        const { status, state: wdsUrlRoot } = await loadWdsUrl()
+        if (status === 'Unauthorized') {
+          notify('warn', 'Error loading data tables', { detail: 'Service returned Unauthorized error. Session might have expired. Please close the tab and re-open it.' })
+        } else if (!!wdsUrlRoot) {
+          if (isLoadRecordTypes) await loadRecordTypes(wdsUrlRoot)
+          await loadRecordsData(recordType, wdsUrlRoot)
+        } else {
+          const errorDetails = await (wdsUrlRoot instanceof Response ? wdsUrlRoot.text() : wdsUrlRoot)
+          // to avoid stacked warning banners due to auto-poll for WDS url, we remove the current banner at 29th second
+          notify('warn', 'Error loading data tables', { detail: `Data Table app not found. Will retry in 30 seconds. Error details: ${errorDetails}`, timeout: WdsPollInterval - 1000 })
+        }
+      } else {
+        // if we have the WDS proxy URL load the WDS data
+        const wdsUrlRoot = wdsProxyUrl.state
+        if (isLoadRecordTypes) await loadRecordTypes(wdsUrlRoot)
+        await loadRecordsData(recordType, wdsUrlRoot)
+      }
+    } catch (error) {
+      notify('error', 'Error loading data tables', { detail: await (error instanceof Response ? error.text() : error) })
+    }
+  }, [loadRecordsData, loadRecordTypes, loadWdsUrl, wdsProxyUrl])
 
   useOnMount(() => {
-    loadTablesData()
     loadRunSet().then(runSet => {
+      setRunSetRecordType(runSet.record_type)
       loadMethodsData(runSet.method_id, runSet.method_version_id)
-      loadRecordsData(runSet.record_type)
+      loadWdsData(runSet.record_type, true)
```

Comment:
I'm not sure what the best practice is here, but I find this second boolean argument a little unintuitive. Does it make sense to have this function accept an object as its argument, so that there can be some indication of what the function is doing at the place in the code where it's being called?

I'm imagining something like: 
`loadWdsData({ recordType: runSet.record_type, includeLoadRecordTypes: true })` 

(this might be a bad idea for some reason that a JS expert would identify...)

Reply:
I don't think that's a bad practice. I have seen usage of passing objects/named parameters to functions as compared to positional arguments in Terra UI (as well as CBAS UI?). Since it improves readability of the code I will update the function to use named parameters 👍